src/Idris/Package.idr

```diff
@@ -432,51 +440,101 @@ getParseErrorLoc fname _ = replFC
 -- it if necessary
 runRepl : {auto c : Ref Ctxt Defs} ->
           {auto o : Ref ROpts REPLOpts} ->
-          PkgDesc -> Core ()
-runRepl pkg
+          PkgDesc ->
+          List CLOpt ->
+          Core ()
+runRepl pkg opts
     = do addDeps pkg
          processOptions (options pkg)
+         preOptions opts
          throw (InternalError "Not implemented")
 
 processPackage : {auto c : Ref Ctxt Defs} ->
                  {auto s : Ref Syn SyntaxInfo} ->
                  {auto o : Ref ROpts REPLOpts} ->
-                 PkgCommand -> String -> Core ()
-processPackage cmd file
-    = do Right (pname, fs) <- coreLift $ parseFile file
-                                  (do desc <- parsePkgDesc file
-                                      eoi
-                                      pure desc)
-             | Left err => throw (ParseFail (getParseErrorLoc file err) err)
-         pkg <- addFields fs (initPkgDesc pname)
-         case cmd of
-              Build => do [] <- build pkg
-                             | errs => coreLift (exitWith (ExitFailure 1))
-                          pure ()
-              Install => do [] <- build pkg
-                               | errs => coreLift (exitWith (ExitFailure 1))
-                            install pkg
-              Clean => clean pkg
-              REPL => runRepl pkg
-
-rejectPackageOpts : List CLOpt -> Core Bool
-rejectPackageOpts (Package cmd f :: _)
-    = do coreLift $ putStrLn ("Package commands (--build, --install, --clean, --repl) must be the only option given")
-         pure True -- Done, quit here
-rejectPackageOpts (_ :: xs) = rejectPackageOpts xs
-rejectPackageOpts [] = pure False
+                 PkgCommand ->
+                 String ->
+                 List CLOpt ->
+                 Core ()
+processPackage cmd file opts
+    =  if not (isSuffixOf ".ipkg" file)
+         then do coreLift $ putStrLn ("Packages must have an '.ipkg' extension: " ++ show file ++ ".")
+                 coreLift (exitWith (ExitFailure 1))
+         else do Right (pname, fs) <- coreLift $ parseFile file
+                                          (do desc <- parsePkgDesc file
+                                              eoi
+                                              pure desc)
+                     | Left (FileFail err) => throw (FileErr file err)
+                     | Left err => throw (ParseFail (getParseErrorLoc file err) err)
+                 pkg <- addFields fs (initPkgDesc pname)
+                 case cmd of
+                      Build => do [] <- build pkg opts
+                                     | errs => coreLift (exitWith (ExitFailure 1))
```

Comment:
I was surprised to see we are not doing anything with the errors but
it turns out that `buildMod` has already emitted all the warnings &
errors. So answering my own question: these are bubbling out but we
do not need to do anything with them.